--- a/configure
+++ b/configure
@@ -1491,6 +1491,10 @@ if test "x$host_alias" != x; then
   fi
 fi
 
+if test "${with_cross_tools+set}" = set; then
+  cross_compiling=yes
+fi
+
 ac_tool_prefix=
 test -n "$host_alias" && ac_tool_prefix=$host_alias-
 
@@ -28661,6 +28665,12 @@ fi
 krb_cv_compile_et="no"
 krb_cv_com_err_need_r=""
 krb_cv_compile_et_cross=no
+
+if test "${with_cross_tools+set}" = set; then
+  krb_cv_compile_et_cross=yes
+  krb_cv_com_err=yes
+fi
+
 if test "${COMPILE_ET}" != "no"; then
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether compile_et has the features we need" >&5
